slide_analysis/descriptors/histogram_descriptor_class.py

```diff
@@ -1,17 +1,14 @@
-from slide_analysis.descriptors.descriptor_class import Descriptor
+import numpy
 
 
-class HistogramDescriptor(Descriptor):
-    def __init__(self, tile, scheme):
-        Descriptor.__init__(self, tile)
-        self.scheme = scheme
-        (self.r_mod, self.g_mod, self.b_mod) = self.scheme
+class HistogramDescriptor:
```

Comment:
class should be either stateful (with constructor method including `tile` and `scheme` and having `self.tile`,`self.value` with method `calc` that fills the `self.value`) or stateless with constructor including only `scheme` and method `calc` which doesn't set properties in object but only returns the `calc` method's `value` by `tile`
```python
class HistogramDescriptor:
    def __init__(self, scheme):
        (self.r_mod, self.g_mod, self.b_mod) = scheme

    def calc(self, tile):
        
        arr = numpy.array(tile.data)
        value = numpy.histogram((arr[:, 0] >> (8 - self.r_mod) << (8 - self.r_mod))
                                     + (arr[:, 1] >> (8 - self.b_mod) << self.g_mod)
                                     + (arr[:, 2] >> (8 - self.g_mod)), bins=numpy.arange(0, 256))[0]
        return value
```
or
```python
class HistogramDescriptor:
    def __init__(self, scheme, tile):
        (self.r_mod, self.g_mod, self.b_mod) = scheme
        self.tile = tile

    def calc(self):
        
        arr = numpy.array(self.tile.data)
        self.value = numpy.histogram((arr[:, 0] >> (8 - self.r_mod) << (8 - self.r_mod))
                                     + (arr[:, 1] >> (8 - self.b_mod) << self.g_mod)
                                     + (arr[:, 2] >> (8 - self.g_mod)), bins=numpy.arange(0, 256))[0]
```